File: backend/internal/domain/saved_place.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type SavedPlaceType string

const (
	SavedPlaceTypeHome  SavedPlaceType = "home"
	SavedPlaceTypeWork  SavedPlaceType = "work"
	SavedPlaceTypeOther SavedPlaceType = "other"
)

type SavedPlace struct {
	ID        uuid.UUID      `json:"id"`
	UserID    uuid.UUID      `json:"user_id"`
	Name      string         `json:"name"`
	Address   string         `json:"address"`
	Latitude  float64        `json:"latitude"`
	Longitude float64        `json:"longitude"`
	Type      SavedPlaceType `json:"type"`
	CreatedAt time.Time      `json:"created_at"`
}

File: mobile/passenger/lib/features/saved_places/models/saved_place.dart
enum SavedPlaceType { home, work, other }

class SavedPlace {
  final String id;
  final String name;
  final String address;
  final double latitude;
  final double longitude;
  final SavedPlaceType type;
  final DateTime? createdAt;

  SavedPlace({
    required this.id,
    required this.name,
    required this.address,
    required this.latitude,
    required this.longitude,
    this.type = SavedPlaceType.other,
    this.createdAt,
  });

  factory SavedPlace.fromJson(Map<String, dynamic> json) {
    return SavedPlace(
      id: json['id'] as String,
      name: json['name'] as String,
      address: json['address'] as String,
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      type: _parseType(json['type'] as String?),
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : null,
    );
  }

  static SavedPlaceType _parseType(String? type) {
    switch (type) {
      case 'home':
        return SavedPlaceType.home;
      case 'work':
        return SavedPlaceType.work;
      case 'other':
      default:
        return SavedPlaceType.other;
    }
  }
}
